Create boards import-data.rst (#5386)
mattermost/docs: source/boards/data-and-archives.rst

Import your data
================

You can import data from other tools to use with Boards.

> Import from Asana

Author: @sbishel, before I continue with the rest of the sections, I want to just make sure the Asana section is correct.

-----------------

This node app converts an Asana JSON archive into a ``.focalboard`` archive.

1. Log into your Asana account.

> 2. From the Asana Board Menu, select **Export/Print > JSON** to create an archive file (which you'll use in Boards).

Author: I wasn't able to find this option in my Asana account. Where is it located?

Reviewer: I had trouble finding it originally as well. I added this to the readme.
From the Asana Board Menu (dropdown next to board title), select Export / Print, and JSON

<img width="720" alt="Screen Shot 2022-02-01 at 10 22 22 AM" src="https://user-images.githubusercontent.com/12704875/152018237-b1abd569-d1bf-4495-a506-43fc60771873.png">

> 2. From the Asana Board Menu, select **Export/Print > JSON** to create an archive file (which you'll use in Boards).

Author: ```suggestion
2. Select the drop-down menu next to the Asana board's name. Then select **Export/Print > JSON**. This will create an archive file which you'll use in Boards.
```

3. Save the file locally, e.g. to ``asana.json``.

> 4. In your local copy of Boards, open a terminal window.

Author: Is this correct? If not, where and how would the user access terminal to enter this command?

Reviewer: Sort of...it really isn't "In you local copy of Boards". (You can probably drop that portion.) But it's just opening a window to run a command prompt. Would be different depending on OS. Mac you can run `terminal`. I think windows is `cmd`. There are also 3rd party packages for this.

> 4. In your local copy of Boards, open a terminal window.

Author: ```suggestion
4. Open a terminal window on your local machine.
```

5. Navigate to ``focalboard/webapp``.
6. Run ``npm install``.
7. Change directory to ``focalboard/import/asana``.
8. Run ``npm install``.

> 9. Run ``npx ts-node importAsana.ts -i <asana.json> -o archive.focalboard``.

Author: Is this still in ``focalboard/import/asana``? Is there any expected output?

Reviewer: Yes that is the correct path.  We display the Board and Card titles that get imported. As well as a summary.
```
Scotts-MacBook-Pro:asana sbishel$ npx ts-node importAsana.ts -i ~/Downloads/asana.json -o archive.focalboard
Board: 1:1 Meeting Agenda Test
Card: [READ ME] Instructions for using this project
Card: [EXAMPLE TASK] Feedback on design team presentation
Card: [EXAMPLE TASK] Finalize monthly staffing plan
Card: [EXAMPLE TASK] Review Q2 launch video outline
Card: [EXAMPLE TASK] Mentor a peer

Found 5 card(s).
Exported to archive.focalboard

```

> 10. Open Boards and select **Settings > Import archive** and select ``archive.focalboard``.

Author: Is this correct? If so, why isn't the user selecting the asana JSON file? If the steps above have changed that, we should probably note it here.

Reviewer: Yes this is still correct. The above command (step 9)...converts `asana.json` to `archive.focalboard` Essentially all of these `imports` are actually conversions to our import format.  We then import our file format.

Reviewer: Prior to Step 10, we probably want to have them open the board in which they want to import into.

11. Select **Upload**.

> 12. Return to your board and confirm that your Asana data is now displaying.

Author: Is this the expected conclusion? What happens if they don't see their Asana data?

Reviewer: Yes. The board will be added to the sidebar. If they don't see their Asana data, the error should be displayed but they should also check log files for errors.

    Card: [EXAMPLE TASK] Finalize monthly staffing plan
    Card: [EXAMPLE TASK] Review Q2 launch video outline
    Card: [EXAMPLE TASK] Mentor a peer
    
    Found 5 card(s).
    Exported to archive.focalboard

9. In Boards, open the board you want to use for the export.
10. Select **Settings > Import archive** and select ``archive.focalboard``.
11. Select **Upload**.
12. Return to your board and confirm that your Asana data is now displaying.

If you don't see your Asana data, an the error should be displayed. You can also check log files for errors.

Import scope
^^^^^^^^^^^^

Currently, the script imports all cards from a single board, including their section (column) membership, names, and notes.

Import from Notion
------------------

This node app converts a Notion CSV and markdown export into a ``.focalboard`` archive.

1. From a Notion Board, open the **...** menu at the top right corner of the board.
2. Select `Export` and pick `Markdown & CSV` as the export format.
3. Save the generated file locally, and unzip the folder.
4. Open a terminal window on your local machine and navigate to ``focalboard/webapp``.
5. Run ``npm install``.

> 6. Change directory to `focalboard/import/notion`.

Author: ```suggestion
6. Change directory to ``focalboard/import/notion``.
```

5. Run ``npm install``.

> 6. Run ``npx ts-node importNotion.ts -i <path to the notion-export folder> -o archive.focalboard``. This generates the following data:

Author: @sbishel, is it possible to provide the same generated data as you did for Asana, for the rest of the tools? If it's the same across the board (barring the tool name) let me know and I can replicate/edit as needed. :)

Import your data
================

You can import data from other tools to use with Boards. [Contribute code](https://www.focalboard.com/contribute/getting-started/) to expand this.

Import from Asana
-----------------

> This node app converts an Asana JSON archive into a ``.focalboard`` archive.

Reviewer 2: Question: Is .focalboard the new or old archive format?

Author: That's a great question. @sbishel - should this be ``.focalboard`` or ``.boardarchive``?

Author: @sbishel, ping on this. :)

> This node app converts an Asana JSON archive into a ``.focalboard`` archive.

Author: ```suggestion
This node app converts an Asana JSON archive into a ``.boardarchive`` file.
```
